src/components/vanilla/charts/BubbleChart/index.tsx

```diff
@@ -0,0 +1,227 @@
+  : (props.isTimeDimension ? data[data.length - 1]?.r : data?.[0]?.r);
+
+  //top padding based on size of top vertical item
+  const highestItem = data.reduce((max, current) => (current?.y > max?.y ? current : max), data[0]);
+  const highestItemRadius = highestItem?.r || 0;
+
+  const yAxisContainsFractions = data?.some(row => !Number.isInteger(row.y));
+
+  return {
+    responsive: true,
+    maintainAspectRatio: false,
+    animation: {
+      duration: 400,
+      easing: 'linear'
+    },
+    scales: {
+      y: {
+        min: props.yAxisMin,
+        grid: {
+          display: false
+        },
+        ticks: {
+          stepSize: !yAxisContainsFractions ? 1 : undefined,
+          padding: firstItemRadius          
+        },
+        title: {
+          display: !!props.yAxisTitle,
+          text: props.yAxisTitle
+        }
+      },
+      x: {
+        reverse: props.reverseXAxis,
+        type: props.isTimeDimension ? 'time' : 'category',
+        time: {
+          round: props.granularity,
+          isoWeekday: true,
+          displayFormats: DATE_DISPLAY_FORMATS,
+          unit: props.granularity
+        },
+        grid: {
+          display: false
+        },
+        title: {
+          display: !!props.xAxisTitle,
+          text: props.xAxisTitle
+        }
+      }
+    },
+    layout: {
+      padding: {
+        left: 0,
+        right: 0,
+        top: highestItemRadius + (props.showLabels ? 30 : 0),
+        bottom: 0
+      }
+    },
+    plugins: {
+      datalabels: {
+        display: props.showLabels ? 'auto' : false,
+        anchor: 'end',
+        align: 'end',
+        font: {
+          weight: 'normal'
+        },
+        formatter: (_, {dataIndex}) => {
+          const v = updatedData?.[dataIndex][props.bubbleSize.name] || 0;
+          const val = formatValue(v, { type: 'number', dps: props.dps || 0 }) || null;
+          return val;
+        }
+      },
+      tooltip: {
+        //https://www.chartjs.org/docs/latest/configuration/tooltip.html
+        callbacks: {
+          label: function (context) {
+            //xAxis label
+            const yAxisLabel = context.dataset.label || '';
+            const rawData = context.raw as { y: number };
+            const yAxisValue = formatValue((`${rawData.y || ''}`), {
+                type: 'number',
+                dps: props.dps
+              });
+            //Bubble size label
+            const bubbleSizeLabel = props.bubbleSize.title;
+            const bubbleSizeValue = updatedData?.[context.dataIndex][props.bubbleSize.name] || 0;
+            const bubbleSizeValueFormatted = formatValue((bubbleSizeValue), {
+                type: 'number',
+                dps: props.dps
+              });            
+            return [`${yAxisLabel}: ${yAxisValue}`, `${bubbleSizeLabel}: ${bubbleSizeValueFormatted}`];
+          }
+        }
+      },
+      legend: {
+        display: !!props.showLegend,
```

Comment:
This might be one of those obscure JS things that I just haven't happened to run across, but any particular reason for the double-negative instead of just doing `display: props.showLegend`?

Reply:
@cwbuecheler the double negative just ensures it's a boolean. Thinking back, I can't remember why exactly I added it like this - it works perfectly with just props.showLegend. I think this is simpler so I'll push a change now.